File: go/src/26.go
func removeDuplicates(nums []int) int {
	if len(nums) < 1 {
		return 0
	}
	if len(nums) == 1 {
		return 1
	}

	length := 1
	for i := 1; i < len(nums); i++ {
		if nums[i] != nums[i-1] {
			nums[length] = nums[i]
			length++
		}
	}

	return length
}

File: py3/26.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0
        elif len(nums) == 1:
            return 1
        
        i = 0  # slow pointer
        for j in range(1, len(nums)):  # fast pointer
            if nums[i] != nums[j]:
                i = i + 1
                nums[i] = nums[j]
        
        return i + 1